lib/auth.guard.ts

```diff
@@ -72,11 +73,11 @@ function createAuthGuard(type?: string | string[]): Type<CanActivate> {
       );
     }
 
-    handleRequest(err, user, info, context, status): TUser {
+    handleRequest<TUser = any>(err, user, info, context, status): TUser {
       if (err || !user) {
         throw err || new UnauthorizedException();
       }
-      return user;
+      return user as any as TUser;
     }
```

Comment:
Can you please revert back changes unrelated to this PR (move the type argument, generic parameter)? Otherwise, it looks good!

Reply:
Hi, thanks for your feedback. I did that to be able to compile. I got an error without this change. But I'll try to do something better :)

Reply:
That should be good now :)